core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -827,8 +827,15 @@ public synchronized void startQueuedRebalance() {
          return;
       }
 
+      // We don't have a queued rebalance.
       if (queuedRebalanceMembers == null) {
-         // We don't have a queued rebalance. We may need to broadcast a stable topology update
+         // The previous topology was not restored. We do nothing, waiting for the members to get back in and install topology.
+         if (currentTopology == null && persistentState.isPresent()) {
+            log.debugf("Skipping rebalance for cache %s as the previous topology was not restored", cacheName);
```

Comment:
I'm now getting stuck at this message and all operations on the cache return:

```
ISPN000689: Recovering cache 'persistentCache' but there are missing members, known members [test-upgrade-1-64083] of a total of 2
```
In both sets of logs on the Jira, the cluster receives a MERGED view from JGroups. The reproducers don't include this.

```
test-upgrade-0 infinispan 14:20:09,108 INFO  (jgroups-6,test-upgrade-0-13791) [org.infinispan.CLUSTER] ISPN000093: Received new, MERGED cluster view for channel test-upgrade: MergeView::[test-upgrade-1-60105|3] (2) [test-upgrade-1-60105, test-upgrade-0-13791], 2 subgroups: [test-upgrade-1-60105|1] (2) [test-upgrade-1-60105, test-upgrade-0-13791], [test-upgrade-1-60105|2] (1) [test-upgrade-1-60105]
test-upgrade-1 infinispan 14:20:09,109 INFO  (jgroups-6,test-upgrade-1-60105) [org.infinispan.CLUSTER] ISPN000093: Received new, MERGED cluster view for channel test-upgrade: MergeView::[test-upgrade-1-60105|3] (2) [test-upgrade-1-60105, test-upgrade-0-13791], 2 subgroups: [test-upgrade-1-60105|1] (2) [test-upgrade-1-60105, test-upgrade-0-13791], [test-upgrade-1-60105|2] (1) [test-upgrade-1-60105]
```

Reply:
We should be covering that with the latest commit :pray: :pray: